### PR TITLE
[Chore](workflow) add clang-tidy workflow

### PR DESCRIPTION
**Pull request by BiteTheDDDDt**
# Proposed changes

testing

## Problem summary

Describe your changes.

## Checklist(Required)

1. Does it affect the original behavior: 
    - [ ] Yes
    - [ ] No
    - [ ] I don't know
2. Has unit tests been added:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
3. Has document been added or modified:
    - [ ] Yes
    - [ ] No
    - [ ] No Need
4. Does it need to update dependencies:
    - [ ] Yes
    - [ ] No
5. Are there any changes that cannot be rolled back:
    - [ ] Yes (If Yes, please explain WHY)
    - [ ] No

## Further comments

If this is a relatively large or complex change, kick off the discussion at [dev@doris.apache.org](mailto:dev@doris.apache.org) by explaining why you chose the solution you did and what alternatives you considered, etc...

### CODE REVIEW

File: .github/workflows/clang-format.yml [hunk 1]
```diff
@@ -17,7 +17,7 @@
 #
 
 ---
-name: BE Code Formatter
+name: BE Code Checks
 on:
   pull_request:
```
Comment (adonis0147) [inline .github/workflows/clang-format.yml:22]: Use `pull_request_target` to enable comment feature.
```suggestion
  pull_request_target
```

File: .github/workflows/clang-format.yml [hunk 2]
```diff
@@ -54,3 +54,57 @@ jobs:
       - name: Ignore it!
         if: ${{ steps.filter.outputs.be_changes == 'false' }}
         run: echo 'No need to check.'
+
+  clang-tidy:
+    name: "Clang Tidy"
+    runs-on: ubuntu-20.04
```
Comment (adonis0147) [inline .github/workflows/clang-format.yml:60]:
```suggestion
    runs-on: ubuntu-22.04
```

File: build.sh [hunk 3]
```diff
@@ -282,6 +282,10 @@ if [[ -z "${USE_DWARF}" ]]; then
     USE_DWARF='OFF'
 fi
 
+if [[ -z "${OUTPUT_BE_BINARY}" ]]; then
+    BUILD_BE_BINARY=1
+fi
+
```
Comment (adonis0147) [inline build.sh]: We don't need this.

File: .github/workflows/clang-format.yml [hunk 4]
```diff
@@ -54,3 +54,57 @@ jobs:
...
+          sudo dpkg-reconfigure --frontend noninteractive tzdata
+
+          pushd thirdparty
+          curl -L https://github.com/apache/doris-thirdparty/releases/download/automation/doris-thirdparty-prebuilt-linux-x86_64.tar.xz \
+            -o doris-thirdparty-prebuilt-linux-x86_64.tar.xz
+          tar -xvf doris-thirdparty-prebuilt-linux-x86_64.tar.xz
+          popd
+
+          export PATH="${DEFAULT_DIR}/ldb-toolchain/bin/:$(pwd)/thirdparty/installed/bin/:${PATH}"
+          DORIS_TOOLCHAIN=clang OUTPUT_BE_BINARY=0 ./build.sh --be
```
Comment (adonis0147) [inline .github/workflows/clang-format.yml:102]: Just using command `cmake </path/contains/CMakeLists.txt>` can generate `compile_commands.json`.

```shell
mkdir build
pushd build
cmake ../be
popd
```

**Review by adonis0147 — CHANGES_REQUESTED** [review]

**Comment by hello-stephen** [comment]
TeamCity pipeline, clickbench performance test result:
 the sum of best hot time: 34.68 seconds
 load time: 430 seconds
 storage size: 17123343075 Bytes
 https://doris-community-test-1308700295.cos.ap-hongkong.myqcloud.com/tmp/20221201042536_clickbench_pr_55893.html

**Pull request closed without merging**